crates/polars-io/src/json/mod.rs

```diff
@@ -154,6 +161,106 @@ where
     }
 }
 
+pub struct JsonBatchedWriter<W: Write> {
+    writer: W,
+    is_first_row: bool,
+    json_format: JsonFormat,
+}
+
+impl<W> JsonBatchedWriter<W>
+where
+    W: Write,
+{
+    pub fn new(writer: W, json_format: JsonFormat) -> Self {
+        JsonBatchedWriter {
+            writer,
+            is_first_row: true,
+            json_format,
+        }
+    }
+}
+
+impl<W> JsonBatchedWriter<W>
+where
+    W: Write,
+{
+    /// Write a batch to the json writer.
+    ///
+    /// # Panics
+    /// The caller must ensure the chunks in the given [`DataFrame`] are aligned.
+    pub fn write_batch(&mut self, df: &DataFrame) -> PolarsResult<()> {
+        let fields = df.iter().map(|s| s.field().to_arrow()).collect::<Vec<_>>();
+        let chunks = df.iter_chunks();
+        match self.json_format {
+            JsonFormat::Json => {
+                self.write_json_batch(fields, chunks)?;
+            },
+            JsonFormat::JsonLines => {
+                self.write_json_lines_batch(fields, chunks)?;
+            },
+        }
+        Ok(())
+    }
+
+    /// Writes the footer of the Json file.
+    pub fn finish(&mut self) -> PolarsResult<()> {
+        if JsonFormat::Json == self.json_format {
+            self.write_json_finish()?;
+        }
+        Ok(())
+    }
+}
+/// These are the methods implementation for json lines format
+impl<W> JsonBatchedWriter<W>
```

Comment:
Hi @svaningelgem,

We decided to take this approach in order to:

- keeping _write_batch_ method small and more readable.
- having two separate and clear implementations (standard json and json lines).
- having two clear _finish_ methods if necessary (currently just for standard json).
- It is easier to identify the code for the implementation of standard json and json lines respectively.

Reply:
I think I didn't make myself clear.

What I'm trying to say is :
```rust
impl<W> JsonBatchedWriter<W>
where
    W: Write,
{
fn write_batch [..]
fn finish [..]
}

impl<W> JsonBatchedWriter<W>
where
    W: Write,
{
fn write_json_lines_batch [..]
}

impl<W> JsonBatchedWriter<W>
where
    W: Write,
{
fn write_json_batch [..]
fn write_json_finish [..]
}
```

All I was trying to convey is to write it like this:
```rust
impl<W> JsonBatchedWriter<W>
where
    W: Write,
{
fn write_batch [..]
fn finish [..]
fn write_json_lines_batch [..]
fn write_json_batch [..]
fn write_json_finish [..]
}
```

When writing this answer, I think it might be even better to write it like this (pseudocode):
```rust
impl<W> JsonBatchedWriter<W>
where
    W: Write,
{
fn _write_batch [..]
fn _finish [..]
}

impl<W> JsonLinesBatchedWriter<W>
where
    W: Write,
{
fn _write_batch [..]
fn _finish [..]
}

impl SinkWriter for JsonBatchedWriter {}
impl SinkWriter for JsonLinesBatchedWriter {}
```

This second approach would allow you to:
- keep the write_batch methods small and readable.
- have two separate and clear implementations (standard json and json lines).
- have a clear finish method.
- easily identify the code for the implementation of standard json and json lines respectively.


WDYT?